### PR TITLE
New package: Sigrok.PulseView.Nightly version nightly

### LINKED ISSUE #162354: [Update Request]: PulseView (Nightly)
### What type of update are you requesting?

A new version of an existing package

### Current Package Identifier

Sigrok.PulseView

### Package Version

nightly

### Please describe the changes you would like to see

The old installer got removed. Here is the new one: https://sigrok.org/download/binary/pulseview/pulseview-NIGHTLY-x86_64-release-installer.exe

### PR DESCRIPTION
**Pull request by schembriaiden**
### Pull request has been created with [komac](https://github.com/russellbanks/Komac) v2.3.0 :rocket:

Resolves: https://github.com/microsoft/winget-pkgs/issues/162354

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/162671)

### CODE REVIEW

**Comment by vikingnope** [comment]
Its probably for the 32 bit app, let me add it

**Comment by vikingnope** [comment]
@stephengillie, you can check again now :)

**Comment by stephengillie** [comment]
> Hi @vikingnope,
> 
> One of the installed EXEs has a dependency on ``Microsoft.VCRedist.2010.` - I'm guessing it's architecture dependent.
> 
> image
> 
> Verify remaining applications: image

Unfortunately, the situation is unchanged. It might be architecture-specific.

**Comment by vikingnope** [comment]
@stephengillie, try again now please :D

**Review by Trenly — COMMENTED** [review]
If this package truly does change nightly, then users will not be able to install it 90% of the time since the hash will be mismatched and by the time the PR is raised and approved to fix the mismatch it will be outdated again in only a few hours

**Comment by Trenly** [comment]
[Policy] Needs Author Feedback

**Comment by stephengillie** [comment]
> If this package truly does change nightly, then users will not be able to install it 90% of the time since the hash will be mismatched and by the time the PR is raised and approved to fix the mismatch it will be outdated again in only a few hours

This is basically an argument against hosting `Nightly` packages, and saying that these in general aren't a good fit for this package manager, due to our pipeline requirements. Is there a better or more dynamic way to handle hash mismatches?

**Comment by Trenly** [comment]
> > If this package truly does change nightly, then users will not be able to install it 90% of the time since the hash will be mismatched and by the time the PR is raised and approved to fix the mismatch it will be outdated again in only a few hours
> 
> This is basically an argument against hosting `Nightly` packages, and saying that these in general aren't a good fit for this package manager, due to our pipeline requirements. Is there a better or more dynamic way to handle hash mismatches?

Yes, that is my argument, but mostly because of the time it takes between the nightly being released and a PR being raised. I don't know that there is a good way to handle nightlies. The only real way for nightlies to be effectively managed is for their PRs to be automated by the publisher's CI/CD, meaning that as long as they pass the pipelines and moderation then they would be merged.

However, even then the nightly may not pass the pipelines on the first try, there may not be a moderator who checks the PR for a few hours, or it could even fail moderation if changes are needed (not likely if it's through CI/CD but there could still be errors). I'd estimate at best a nightly would be available for around 18 hours a day if nothing goes wrong, which is still only 75% availability for the package

**Comment by stephengillie** [comment]
> Unfortunately, the situation is unchanged.

**Comment by vikingnope** [comment]
I removed x86 and the dependencies as it was before. It should be fine now if you test it.

File: manifests/s/Sigrok/PulseView/Nightly/nightly/Sigrok.PulseView.Nightly.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+PackageIdentifier: Sigrok.PulseView.Nightly
+PackageVersion: nightly
+InstallerType: nullsoft
+InstallModes:
+- silent
+- silentWithProgress
+UpgradeBehavior: install
+ReleaseDate: 2024-03-09
+Installers:
```
Comment (stephengillie) [inline manifests/s/Sigrok/PulseView/Nightly/nightly/Sigrok.PulseView.Nightly.installer.yaml:12]:
```suggestion
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2010.x64
Installers:
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
With this dependency, the package installed and launched normally.
image

**Comment by stephengillie** [comment]
Verify:
image

**Comment by vikingnope** [comment]
I will check about the version later

**Review by stephengillie — APPROVED** [review]

**Comment by stephengillie** [comment]
I meant to approve this earlier. Sorry for the delay.

**Pull request merged**